### ISSUE TITLE
Background task for printer fetching is slower than expected

### ISSUE THREAD

**Issue by tresf** [qzind/tray#925]
Today, a client with ~1,200 printers was able to illustrate that with QZ Tray 2.1.5, the startup task which fetches the printer details does not properly cache them.

This can be observed as follows:

✅  Initial background task time: ~3 minutes
🚫 First search for printers after background task is completed: ~1 minute
✅ Second search for printers: 2 seconds

@bberenz can you see if we have something awry with our printer searching logic?

**Comment by klabarge** [comment]
I'm having a hard time reproducing this bug. I created 2011 printers in a VM. It took about 12 minutes for the `IDLE` task to complete. However the first printer search after this task took less than 1 second.

**On Startup**
```
[DEBUG] 2022-03-03T21:07:34,499 @ qz.common.TrayManager:-1
        IDLE: Performing first run of find printers
[INFO] 2022-03-03T21:08:21,192 @ qz.installer.certificate.CertificateManager:-1
        SSL certificate is still valid for 824 more days: 2024-06-04T22:06Z.  We'll make a new one automatically when needed.
[DEBUG] 2022-03-03T21:19:01,576 @ qz.printer.PrintServiceMatcher:-1
        Found 2011 printers
```

**First Use**
```
[DEBUG] 2022-03-03T21:22:38,370 @ qz.printer.PrintServiceMatcher:-1
        Searching for PrintService matching Test Printer 1000
[DEBUG] 2022-03-03T21:22:38,584 @ qz.printer.PrintServiceMatcher:-1
        Found 2011 printers
[DEBUG] 2022-03-03T21:22:38,589 @ qz.printer.PrintServiceMatcher:-1
        Found match: Test Printer 1000
```

**Comment by tresf** [comment]
The patch provided in #929 avoids looping over Media types (such as printer trays) when details are not requested. If there's a way to simulate this, I think it will be a better test-case.

**Comment by tresf** [comment]
I was able to reproduce the issue using @klabarge's script here: https://gist.github.com/klabarge/46dd9cc2e878ed46b5feff38b169d270

1. I downloaded and installed a fake "Boca BIDI FGL 200 DPI" printer
   a. This specific driver is used because BOCA takes advantage of the "[Trays](https://user-images.githubusercontent.com/6345473/156794106-9b397053-87b4-44ad-8390-230f03a33740.png)" for controlling certain printer features, sending us into the `for(Media media : ...)` loop that this PR fixes.
2. Replaced "GetRandomDriver()" with "Boca BIDI FGL 200 DPI".
3. Changed 2,000 to 100
4. Ran script

What I observed:
* About 30 seconds to initially find printers
* About 10 seconds to gather the tray information

This PR no longer has the second delay, so I consider this patched.

Some future considerations (e.g. master branch), @bberenz this background task deadlocks the UI (can't access the QZ menu while running), is there a better way to do this?  Feel free to open a new issue if you agree.
